contracts/gauge-borrowing/VaultPriceOracle.sol

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/utils/math/SafeMathUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+
+import "../interfaces/Compound/IPriceOracle.sol";
+import "../interfaces/Compound/IBasePriceOracle.sol";
+import "../interfaces/Compound/ICToken.sol";
+import "../interfaces/Compound/ICErc20.sol";
+import "../interfaces/Compound/IAggregatorV3Interface.sol";
+import "../interfaces/Compound/ILiquidityGauge.sol";
+import "../interfaces/Compound/IRibbonVault.sol";
+import "hardhat/console.sol";
+/**
+ * @title VaultPriceOracle
+ * @notice Returns prices from Chainlink.
+ * @dev Implements `PriceOracle`.
+ * @author David Lucid <david@rari.capital> (https://github.com/davidlucid)
+ */
+contract VaultPriceOracle is IPriceOracle, IBasePriceOracle {
+    using SafeMathUpgradeable for uint256;
+
+    /**
+     * @notice Maps ERC20 token addresses to ETH-based Chainlink price feed contracts.
+     */
+    mapping(address => IAggregatorV3Interface) public priceFeeds;
+
+    /**
+     * @notice Maps ERC20 token addresses to enums indicating the base currency of the feed.
+     */
+    mapping(address => FeedBaseCurrency) public feedBaseCurrencies;
+
+    /**
+     * @notice Enum indicating the base currency of a Chainlink price feed.
+     */
+    enum FeedBaseCurrency {
+        ETH,
+        USD,
+        BTC
+    }
+
+    /**
+     * @notice Chainlink ETH/USD price feed contracts.
+     */
+    IAggregatorV3Interface public constant ETH_USD_PRICE_FEED = IAggregatorV3Interface(0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419);
```

Comment:
Better to set these addresses in the construct for multi-chain or testnet deployments because the addresses could be different.

Reply:
Also could use `immutable` to save a bit of gas if we set in the constructor

Reply:
long way until pool on arbitrum so until then can just keep as it and then swap later on